### DISCUSSION TITLE
Google Play Store : You uploaded an APK or Android App Bundle signed with a certificate that expires too soon. You need to sign your APK or Android App Bundle with a certificate that expires farther into the future.

### DISCUSSION THREAD

**Discussion by lwmirkk** [airsdk/Adobe-Runtime-Support#2482] [Q&A] [answered]
Hi!
I'm creating a new app and received this error when upload the AAB.
Checked the certificate and the expire will be on 2027....
The certificate was created in Dec. 2022 with AIR (changed the name to generic names):
`33.1.1.935\bin\adt.bat -certificate -cn "app_name" -o "org_name" -c "COUNTRY_CODE" 2048-RSA keystore.p12 "password"`

Google integrity check is with these settings:
- Releases signed by Google Play
- Google will generate and protect an app signing key for your releases

As this is the second app I sent to Play Store, the first one doen't show this error when I update.

Someone already see this error?

Thanks a lot. 

EDIT:
The first app was sent to the play store before the AAB was mandatory, but it is updated until today.
The certificate of the first app, oddly enough, will expire in 2044 !?
The second app certificate that I cried afterwards in Dec.2022 will expire in 2027...!?

**Comment by FliplineStudios** [comment, marked as answer]
The Play Store requires certificates to be valid at least through October 22, 2033, so that would be why you're getting that error.  When you use the adt certificate command, you should also use the -validityPeriod flag and specify a longer amount of years (Google recommends 25 years), I believe AIR defaults to 5 years which is too short for Google's requirements.

**Reply by ajwfrost** [reply]
Thanks @FliplineStudios  -> do you think it's worth us changing that default of 5 years?
It looks like currently there's a limit on our certificates that they can't go beyond 2049 (because of how rfc2459 defines the utc date fields in certificates) .. but that gives us 20+ months to add the support for the new generic date format... :-)

thanks

**Reply by lwmirkk** [reply]
I think worth changing the default 5 years, Flash CC default certificate expire was about 25 years if I remember.

**Comment by lwmirkk** [comment]
Bingo! Thats the problem, thanks a lot!
=)
